### PR TITLE
[PyROOT experimental] Protect missing declarations for dataframe disabled

### PR DESCRIPTION
**Pull request by stwunsch**
This would be an intermediate fix to make PyROOT experimental build without dataframe enabled. However, this points to an underlying problem that we require the C++ side pythonizations more to be more modularized.

@etejedor Shall we go for this fix for now and go for the proper fix later? Or shall we go for a proper fix directly?

### CODE REVIEW

**Comment by stwunsch** [comment]
@phsft-bot build with `-Dpyroot_experimental=ON`

**Comment by stwunsch** [comment]
@phsft-bot build with -Dpyroot_experimental=ON

**Comment by etejedor** [comment]
If @Axel-Naumann agrees, I am fine with this temporary fix.

**Comment by stwunsch** [comment]
I'll write later a Jira ticket so that we follow up the modularization going towards 6.22.

**Comment by stwunsch** [comment]
There we go: https://sft.its.cern.ch/jira/browse/ROOT-10382

**Comment by stwunsch** [comment]
@phsft-bot build with `-Dpyroot_experimental=ON -Ddataframe=OFF`

File: tmva/tmva/inc/LinkDef1.h [hunk 1]
```diff
@@ -68,8 +70,9 @@
 #pragma link C++ class TMVA::MethodCrossValidation+;
 #pragma link C++ class TMVA::MethodDL+;
 
+#ifdef R__HAS_DATAFRAME
 // BDT inference
 #pragma link C++ class TMVA::Experimental::RBDT<TMVA::Experimental::BranchlessForest<float>>;
```
Comment (hageboeck) [inline tmva/tmva/inc/LinkDef1.h]: It looks like this is a bit too restrictive, and we are creating more depedencies than necessary. As a hotfix, it's probably ok, but can this also work without dataframe?

Comment (stwunsch) [reply tmva/tmva/inc/LinkDef1.h]: In principle yes, however, TMVA experimental has RDF as dependency and therefore we carry this even to some classes, which are actually not dependent on the RDF code itself. The fine granular splitting of the dependencies has a significant overhead and therefore I would go for this solution.

**Pull request merged**